consensus-types/blocks/factory.go

```diff
@@ -29,7 +29,32 @@ var (
 	errNonBlindedSignedBeaconBlock = errors.New("can only build signed beacon block from blinded format")
 )
 
+// NewSignedBeaconBlockWithVersion creates a signed beacon block from a protobuf signed beacon block.
+// This function is needed because Electra & Fulu uses the same protobuf message for blinded and unblinded blocks
+// and we need to handle them differently based on the fork version.
+func NewSignedBeaconBlockWithVersion(ver int, i interface{}) (interfaces.SignedBeaconBlock, error) {
```

Comment:
I'm a little bit back and forth about this, honestly after trying it out with some test fixes, I'd prefer to directly update `NewSignedBeaconBlock` to this signature, instead of adding this new function. Reason being:

1. There could be silent failures that is not covered by tests and can only be discovered in prod (this is a good enough reason already)
2. potential inconsistent api calls in the future, people might still be used to the old `NewSignedBeaconBlock` func

This being said, would love your thoughts on if `NewSignedBeaconBlock(ver int, i any)` make sense